lib/common/strings.c

```diff
@@ -148,6 +150,129 @@ crm_parse_int(const char *text, const char *default_text)
     return (int) result;
 }
 
+/*!
+ * \internal
+ * \brief Scan a double-precision floating-point value from a string
+ *
+ * \param[in]  text      String to scan
+ * \param[out] result    If not NULL, where to store scanned value
+ * \param[out] end_text  If not NULL, where to store pointer to just
+ *                       after value
+ *
+ * \return Standard Pacemaker return code (also set errno on error)
+ */
+static int
+pcmk__scan_double(const char *text, double *result, char **end_text)
+{
+    double local_result = -1.0;
+    char *local_end_text = NULL;
+    int rc = pcmk_rc_ok;
+
+    errno = 0;
+    if (text != NULL) {
+        local_result = strtod(text, &local_end_text);
+
+        if (errno == ERANGE) {
+            /*
+             * Overflow: strtod() returns +/- HUGE_VAL and sets errno to
+             *           ERANGE
+             *
+             * Underflow:
+             * - ANSI: strtod() returns 0 and sets errno to ERANGE
+             * - C99:  strtod() returns "a value whose magnitude is no
+             *         greater than the smallest normalized positive"
+             *         double. Whether ERANGE is set is implementation-
+             *         defined.
+             */
+            rc = errno;
+            crm_warn("Floating-point value parsed from '%s' would %sflow "
+                     "(using %g instead)", text,
+                     (fabs(local_result) > DBL_MIN) ? "over" : "under",
+                     local_result);
+
+        } else if (errno != 0) {
+            rc = errno;
+            local_result = -1;
```

Comment:
I don't think so. strtod() sets local_result = 0 if it can't parse the string. This line sets it back to -1.

And the initialization of local_result to -1.0 remains useful in the event that (text == NULL) and we don't enter this main block. We would have to either initialize to -1.0 as we do now, or set it to -1.0 in an else block (i.e., text == NULL).

Reply:
Whoops, my bad, I missed that.

Reply:
No problem. It's easy to miss, so good thing you brought it up for me to comment the reason for re-setting it.